hrqb/tasks/employee_appointments.py

```diff
@@ -0,0 +1,148 @@
+        emp_appts_df = dw_emp_appts_df.merge(
+            libhr_df[
+                [
+                    "Related Employee MIT ID",
+                    "Position ID",
+                    "Related Supervisor MIT ID",
+                    "HC ID",
+                    "Related Department ID",
+                    "Cost Object",
+                ]
+            ],
+            how="left",
+            left_on=["position_id", "mit_id"],
+            right_on=["Position ID", "Related Employee MIT ID"],
+        )
+
+        # merge on departments to get directorates
+        emp_appts_df = emp_appts_df.merge(
+            depts_df[["Record ID#", "Directorate"]],
+            how="left",
+            left_on="Related Department ID",
+            right_on="Record ID#",
+        )
+
+        emp_appts_df = normalize_dataframe_dates(
+            emp_appts_df,
+            [
+                "appt_begin_date",
+                "appt_end_date",
+            ],
+        )
+
+        fields = {
+            "hr_appt_key": "HR Appointment Key",
+            "mit_id": "MIT ID",
+            "HC ID": "HC ID",
+            "employee_type": "Related Employee Type",
+            "appt_begin_date": "Begin Date",
+            "appt_end_date": "End Date",
+            "Directorate": "Related Directorate",
+            "Related Department ID": "Related Department ID",
+            "Related Supervisor MIT ID": "Supervisor",
+            "job_title_long": "Related Job Title",
+            "position_title_long": "Related Position Title",
+            "job_family": "Job Family",
+            "job_subfamily": "Job Subfamily",
+            "job_track": "Job Track",
+            "position_id": "Position ID",
+            "Cost Object": "Cost Object",
+            "exempt": "Exempt / NE",
+            "union_name": "Union Name",
+            "term_or_perm": "Term or Permanent",
+            "benefits_group_type": "Benefits Group Type",
+        }
+
+        return emp_appts_df[fields.keys()].rename(columns=fields)
+
+
+class LoadEmployeeAppointments(QuickbaseUpsertTask):
+
+    stage = luigi.Parameter("Load")
+    table_name = "Employee Appointments"
+
+    def requires(self) -> list[luigi.Task]:  # pragma: nocover
+        from hrqb.tasks.employee_types import LoadEmployeeTypes
+        from hrqb.tasks.job_titles import LoadJobTitles
+        from hrqb.tasks.position_titles import LoadPositionTitles
+
+        return [
+            LoadEmployeeTypes(pipeline=self.pipeline),
+            LoadJobTitles(pipeline=self.pipeline),
+            LoadPositionTitles(pipeline=self.pipeline),
+            TransformEmployeeAppointments(pipeline=self.pipeline),
+        ]
+
+    @property
+    def merge_field(self) -> str | None:
+        return "HR Appointment Key"
+
+    @property
+    def input_task_to_load(self) -> str:
+        """Upsert data from parent task 'TransformEmployeeAppointments'."""
+        return "TransformEmployeeAppointments"
```

Comment:
This is an example of the using the optional property `input_task_to_load`.  By setting this, it explicitly sets which parent task from the multiple ones defined in `self.requires()` to use as the input data for upserting.